src/exports.cc

```diff
@@ -36,6 +36,24 @@ ContextBase *contextOrEffectiveContext() {
 // of current_context_.
 extern thread_local uint32_t effective_context_id_;
 
+// Lookup table of foreign functions, using a pointer to avoid the initialization fiasco.
+std::unordered_map<std::string, WasmForeignFunction> *foreign_functions = nullptr;
+
+WasmForeignFunction getForeignFunction(std::string_view function_name) {
+  auto it = foreign_functions->find(std::string(function_name));
```

Comment:
I think you need to add guard here, otherwise you might be dereferencing `nullptr`.

Reply:
https://github.com/proxy-wasm/proxy-wasm-cpp-host/pull/190/commits/7893fc75855a0057741b71917224860186037d45 done!